src/Boids.hs

```diff
@@ -173,12 +176,25 @@ zoomControl (EventKey (MouseButton WheelDown) Down _ _) world =
     world { worldViewPort = zoom (-zoomSpeed) $ worldViewPort world }
 zoomControl _ world = world
 
--- TODO: implement dragging around
+dragControl :: Event -> World -> World
```

Comment:
Those `*Control` function actually work only with `ViewPort`. They don't need the entire world. Reflect that in the signature and refactor accordingly.